packages/builder/src/pages/builder/portal/settings/ai/index.svelte

```diff
@@ -0,0 +1,171 @@
+<script>
+  import { onMount } from "svelte"
+  import {
+    Button,
+    Layout,
+    Heading,
+    Body,
+    Helpers,
+    Divider,
+    notifications,
+    Modal,
+    Tags,
+    Tag,
+  } from "@budibase/bbui"
+  import { admin, licensing } from "stores/portal"
+  import { API } from "api"
+  import AIConfigModal from "./ConfigModal.svelte"
+  import AIConfigTile from "./AIConfigTile.svelte"
+
+  const ConfigTypes = {
+    AI: "ai",
+  }
+
+  let modal
+  let fullAIConfig
+  let editingAIConfig = {}
+  let editingUuid
+
+  $: isCloud = $admin.cloud
+  $: budibaseAIEnabled = $licensing.budibaseAIEnabled
+  $: customAIConfigsEnabled = $licensing.customAIConfigsEnabled
+
+  async function fetchAIConfig() {
+    try {
+      const aiDoc = await API.getConfig(ConfigTypes.AI)
+      if (aiDoc._id) {
+        fullAIConfig = aiDoc
+      }
+    } catch (error) {
+      notifications.error("Error fetching AI config")
+    }
+  }
+
+  async function saveConfig() {
+    // Use existing key or generate new one
+    const id = editingUuid || Helpers.uuid()
+
+    // Creating first custom AI Config
+    if (!fullAIConfig) {
+      fullAIConfig = {
+        type: ConfigTypes.AI,
+        config: {
+          [id]: editingAIConfig,
+        },
+      }
+    } else {
+      // We don't store the default BB AI config in the DB
+      delete fullAIConfig.config.budibase_ai
+      // unset the default value from other configs if default is set
+      if (editingAIConfig.isDefault) {
+        for (let key in fullAIConfig.config) {
+          fullAIConfig.config[key].isDefault = false
+        }
+      }
+      // Add new or update existing custom AI Config
+      fullAIConfig.config[id] = editingAIConfig
+    }
+
+    try {
+      await API.saveConfig(fullAIConfig)
+      notifications.success(`Successfully saved and activated AI Configuration`)
+      await fetchAIConfig()
+    } catch (error) {
+      notifications.error(
+        `Failed to save AI Configuration, reason: ${
+          error?.message || "Unknown"
+        }`
+      )
+    }
```

Comment:
Given that `fetchAIConfig` does its own toast on failure and can never throw, I'd move it out of this try to improve clarity.